### PR TITLE
Stops register page from using deprecated setting

### PR DESCRIPTION
**Pull request by caesar2164**
@stvstnfrd this may not be the final version, but this works for deploy.

### CODE REVIEW

**Comment by caesar2164** [comment]
[CME vs Lagunita vs SUclass Registration settings](https://trello.com/c/lfVxKjjB/201-cme-vs-lagunita-vs-suclass-registration-settings)

**Comment by stvstnfrd** [comment]
> this works for deploy

It doesn't, though. 
As is, when we redeploy stage, this value will revert to `lagunita`.

As discussed, `DEFAULT_SITE_THEME` is, as the name suggests,
only meant to be the _default_ value, used for initialization,
when the value isn't defined in the database.

Currently, this defaults to `lagunita` on _all_ instances.
The only reason this "works" on stage is because you hard-coded `common.py`.

We can't assume that `DEFAULT_SITE_THEME == CURRENT_SITE_THEME`.

There are a number of other ways we could go about this cleanly,
so let's avoid ugly hacks.

The simplest approach is to attempt is to create an optional underscore template
that we pull from our theme, and then have `register.underscore` pull in and render this template.

There are a number of variations we could attempt if there are issues,
but all the documentation I'm seeing indicates that this should be straight-forward.

**Comment by caesar2164** [comment]
Why would **SUclass** and **CME** have any theme other than the **suclass** and **cme** themes respectively set as their `DEFAULT_SITE_THEME`'s?

**Comment by caesar2164** [comment]
@stvstnfrd I haven't seen a good way to have conditional includes in underscore templates?

**Comment by stvstnfrd** [comment]
> I haven't seen a good way to have conditional includes in underscore templates?

@caesar2164 No, the idea is you "optionally" dump an underscore template to the page (either pulled from theme or blank), then in `register.underscore`, you check for the existence of the new "sub-"template, and if it exists, you pull it in too and render.

**Comment by stvstnfrd** [comment]
# `register-extra.underscore`

```
// put the extra contents here
// default is empty/blank, but can be overridden via theme
```

# `register.underscore`

```
var t = template($('#register-extra-template').html());
$('SOME_DIV_IN_MAIN_TEMPLATE').html(t({ n: 17 }));
```

https://stackoverflow.com/a/13649447
https://emptysqua.re/blog/adding-an-include-tag-to-underscore-js-templates/

**Comment by stvstnfrd** [comment]
And really, I don't even think the `register-extra` "sub-template" even needs to be underscore, per-se. It could probably also be a regular Python template that gets dumped/hidden on the page.

File: lms/static/js/student_account/views/RegisterView.js [hunk 1]
```diff
@@ -56,12 +55,14 @@
                             currentProvider: this.currentProvider,
                             providers: this.providers,
                             hasSecondaryProviders: this.hasSecondaryProviders,
-                            themeName: this.themeName,
                             privacyPolicyUrl: this.privacyPolicyUrl,
                             platformName: this.platformName
                         }
                     }));
 
+                    var regExtra = $('#register-extra-tpl').html();
```
Comment (stvstnfrd) [inline lms/static/js/student_account/views/RegisterView.js]: If you still want to make this a one-liner, I'll look the other way ;)

**Comment by stvstnfrd** [comment]
Added a concession, otherwise,
this looks good pending tests and https://github.com/Stanford-Online/openedx-themes/pull/7

File: lms/static/js/student_account/views/RegisterView.js (hunk 1 above)
Comment (caesar2164) [reply lms/static/js/student_account/views/RegisterView.js]: nah, I'm cool as is.

**Pull request merged**